sdk/storage/azblob/lease/blob_client.go

```diff
@@ -102,19 +102,19 @@ func (c *BlobClient) BreakLease(ctx context.Context, o *BlobBreakOptions) (BlobB
 
 // ChangeLease changes the blob's lease ID.
 // For more information, see https://docs.microsoft.com/rest/api/storageservices/lease-blob.
-func (c *BlobClient) ChangeLease(ctx context.Context, o *BlobChangeOptions) (BlobChangeResponse, error) {
+func (c *BlobClient) ChangeLease(ctx context.Context, proposedLeaseID string, o *BlobChangeOptions) (BlobChangeResponse, error) {
```

Comment:
I thought that having an auto-generated lease ID, thus making the proposed lease ID optional, was the desired behavior.  I'm not pushing back on the change as I don't know enough about this scenario, but I want to confirm that this new behavior is what we want.

CC @JeffreyRichter

Reply:
There are times when a customer needs to specify a precise lease ID. But, we could make it optional; not mandatory.

Reply:
Yes agreed, and that's what we previously had implemented.

Reply:
I think part of the original confusion is that, while the proposed lease ID is required by the REST endpoint, we didn't doc that we set it to a random UUID on your behalf if you don't specify your own.  If we are to keep the previous behavior, then we need to doc it.

Reply:
If I remember correctly, the service doesn't "require" this but we always want to pass it because it makes the operation idempotent so we can retry it.

Reply:
Oh, you're right. I thought you were talking about AcquireLease; not ChangeLease. For ChangeLease, you could see the ID being required because if the customer wants to change the lease, they most-likely have a specific lease ID in mind to change it to. Otherwise, there is no real need to change the lease.

Reply:
@jhendrixMSFT @JeffreyRichter yea the new lease id is required for the rest endpoint, as @JeffreyRichter said the point of this is operation is to change the id so it should be precise by the customer anyway. So just to confirm, this change looks ok?